Fix themes on page previews

ice9js opened the pull request:
This is a partial fix for themes not updating on page previews. This patch will force the page navigation column to re-render if the current theme changes.  
This fixes the issue for themes but unfortunately other issues where certain block styles might not load correctly inside the preview. It's also not an ideal experience as it takes a few seconds for all previews to re-render from scratch so it's definitely noticeable.

The core issue is related to Gutenberg itself not allowing the preview styles to be updated once the preview is rendered ([link](https://github.com/WordPress/gutenberg/blob/f2161e246b9fdd9a2a56e7552b0b28050f1a5302/packages/block-editor/src/components/iframe/index.js#L230)).  
A proper fix would involve creating our own Preview component which does allow for this which would allow us to address remaining edge cases as well as provide a much smoother experience.

# Testing

Try changing themes using the theme modal in the editor. Page previews should be updated with the theme styles.

digitalwaveride reviewed (approved):
Nice Kuba! 😻 

Yes, it's not perfect, but for now... a great fix for allowing us to move on!

💪 🙏

ice9js commented:
I added a small fix to make sure the page previews background matches the theme and updated leven to use `--wp-custom-color-background` variable that seems to be the standard for newer themes, as discussed with @jcheringer.

jcheringer reviewed (approved):
LGTM! :shipit:

digitalwaveride commented:
Just found this issue for "Starting from Scratch" template:

![image](https://user-images.githubusercontent.com/5120357/159289199-ddb9930b-08b5-4975-9b88-582df0f88217.png)

ice9js commented:
>Just found this issue on "Start from Scratch" template...

Already been addressed :)
